pkg/minikube/cluster/cluster.go

```diff
@@ -313,6 +342,17 @@ func (m *MachineConfig) CacheMinikubeISOFromURL() error {
 	}
 
 	defer response.Body.Close()
+	isoData, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return err
+	}
+
+	// Validate the ISO if it was the default URL, before writing it to disk.
+	if m.MinikubeISO == constants.DefaultIsoUrl {
+		if !isIsoChecksumValid(&isoData, constants.DefaultIsoShaUrl) {
```

Comment:
Do we just leave the sha itself as a constant instead of the iso sha url?  The SHA will always be the same since we are always requesting the same default iso SHA.  (Maybe that would add extra complexity for the sake of saving an http request)